File: Vachagan/CS50x 2019/Week 6/caesar/caesar.py
from sys import argv
from cs50 import get_string


def main(argv):

    if (len(argv) != 2):
        print("Usage: ./caesar key\n")
        exit(1)

    key = int(argv[1]) % 26
    word = get_string("plaintext: ")
    print("ciphertext: ", end='')
    for leter in word:
        if (leter.isalpha()):
            asciiLeter = 65 if ord(leter) < 97 else 97
            leter = chr((ord(leter) + key - asciiLeter) % 26 + asciiLeter)
        print(leter, end='')
    print()
    exit(0)


if __name__ == "__main__":
    main(argv)

File: Vachagan/CS50x 2019/Week 1 continued/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    int key = atoi(argv[1]) % 26;
    string word = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0; i < strlen(word); i++) 
    {
        if (isalpha(word[i]))
        {
            int ascii = (word[i] < 97) ? 65 : 97;
            word[i] = (char)((word[i] + key - ascii) % 26 + ascii);
        }
        printf("%c", word[i]);
    }

    printf("\n");
    return 0;
}